exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1322 @@
+		})
+	case asset.Option:
+		var instsb []Instrument
+		var instsc []Instrument
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "BTC-USD",
+		})
+		if err != nil {
+			return nil, err
+		}
+		instsb, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "ETH-USD",
+		})
+		if err != nil {
+			return nil, err
+		}
+		instsc, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "SOL-USD",
+		})
+		insts = append(insts, instsb...)
+		insts = append(insts, instsc...)
+	case asset.Margin:
+		insts, err = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeMargin,
+		})
+	}
+	if err != nil {
+		return nil, err
+	}
+	if len(insts) == 0 {
+		return nil, errNoInstrumentFound
+	}
+	for x := range insts {
+		var pair string
+		switch insts[x].InstrumentType {
+		case asset.Spot, asset.Margin:
+			pair = insts[x].BaseCurrency + format.Delimiter + insts[x].QuoteCurrency
+			if pair == "-" {
+				return nil, fmt.Errorf("%v, invalid currency pair data", errMalformedData)
+			}
+		case asset.Futures, asset.PerpetualSwap, asset.Option:
+			c, err := currency.NewPairFromString(insts[x].InstrumentID)
+			if err != nil {
+				return nil, err
+			}
+			pair = c.Base.String() + format.Delimiter + c.Quote.String()
+		}
+		pairs = append(pairs, pair)
+	}
+	selectedPairs := []string{}
+	pairsMap := map[string]int{}
+	for i := range pairs {
+		if pairs[i] == "" {
+			return nil, errInvalidCurrencyPair
+		}
+		count, ok := pairsMap[pairs[i]]
+		if !ok || count == 0 {
+			pairsMap[pairs[i]] = 1
+			selectedPairs = append(selectedPairs, pairs[i])
+		}
+	}
+	return selectedPairs, nil
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores them in the exchanges config
+func (ok *Okx) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	assetTypes := ok.GetAssetTypes(false)
+	for i := range assetTypes {
+		p, err := ok.FetchTradablePairs(ctx, assetTypes[i])
+		if err != nil {
+			return err
+		}
+		pairs, err := currency.NewPairsFromStrings(p)
+		if err != nil {
+			return err
+		}
+		err = ok.UpdatePairs(pairs, assetTypes[i], false, forceUpdate)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// UpdateTicker updates and returns the ticker for a currency pair
+func (ok *Okx) UpdateTicker(ctx context.Context, p currency.Pair, a asset.Item) (*ticker.Price, error) {
+	format, err := ok.GetPairFormat(a, false)
+	if err != nil {
+		return nil, err
+	}
+	if p.Base.String() == "" || p.Quote.String() == "" {
+		return nil, errIncompleteCurrencyPair
+	}
+	instrumentID := format.Format(p)
+	if !ok.SupportsAsset(a) {
+		return nil, fmt.Errorf("%w: %v", asset.ErrNotSupported, a)
+	}
+	mdata, err := ok.GetTicker(ctx, instrumentID)
+	if err != nil {
+		return nil, err
+	}
+	err = ticker.ProcessTicker(&ticker.Price{
```

Comment:
Need to check out why this is dropping the contract code when updating for websocket. 
```
[INFO]  | SYNC | 27/09/2022 10:37:21 | Okx websocket TRX-USDT FUTURES TICKER: Last 0.05714000 Ask 0.05717000 Bid 0.05710000 High 0.05726000 Low 0.05628000 Volume 53669000.00000000
```